Source/Scene/MapboxStyleProvider.js

```diff
@@ -0,0 +1,371 @@
+define([
+    '../Core/Credit',
+    '../Core/defaultValue',
+    '../Core/defined',
+    '../Core/defineProperties',
+    '../Core/DeveloperError',
+    '../Core/MapboxApi',
+    '../Core/Resource',
+    './UrlTemplateImageryProvider'
+], function(
+    Credit,
+    defaultValue,
+    defined,
+    defineProperties,
+    DeveloperError,
+    MapboxApi,
+    Resource,
+    UrlTemplateImageryProvider) {
+'use strict';
+
+var trailingSlashRegex = /\/$/;
+var defaultCredit = new Credit('&copy; <a href="https://www.mapbox.com/about/maps/">Mapbox</a> &copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a> <strong><a href="https://www.mapbox.com/map-feedback/">Improve this map</a></strong>');
+
+/**
+ * Provides tiled imagery hosted by Mapbox.
+ *
+ * @alias MapboxStyleProvider
+ * @constructor
+ *
+ * @param {Object} [options] Object with the following properties:
+ * @param {String} [options.url='https://api.mapbox.com/styles/v1/'] The Mapbox server url.
```

Comment:
This should be a `String` or a `Resource` like all our terrain/imagery providers.

Reply:
@tfili I kept it as `String` in order to be the same as the current `MapboxImageryProvider`. Should I change it?

Reply:
Yeah, all our providers take `Resource` objects now. The MapBox Imagery provider was an oversight.

Reply:
@tfili I updated the type of the parameter. I think the code already takes resource into account 😉